modules/acquia_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,205 @@
+   *   "[Amplitude] Revenue (Unverified)", and "[Amplitude] Merged User".
+   *
+   * @param array $event_properties
+   *   Event properties.
+   *
+   * @throws \Exception
+   *
+   * @see https://amplitude.zendesk.com/hc/en-us/articles/204771828#keys-for-the-event-argument
+   */
+  public function sendTelemetry($event_type, array $event_properties = []) {
+    $event = $this->createEvent($event_type, $event_properties);
+
+    // Failure to send Telemetry should never cause a user facing error or
+    // interrupt a process. Telemetry failure should be graceful and quiet.
+    try {
+      $this->sendEvent($event);
+    }
+    catch (\Exception $e) {
+      if (getenv('ACQUIA_TELEMETRY_LOUD')) {
+        throw $e;
+      }
+    }
+  }
+
+  /**
+   * Get an array of information about Lightning extensions.
+   *
+   * @return array
+   *   An array of extension info keyed by the extensions machine name.
+   */
+  protected function getExtensionInfo() {
+    $all_modules = $this->extensionListModule->getAllAvailableInfo();
+    $acquia_extensions = array_intersect_key($all_modules, array_flip($this->getAcquiaExtensionNames()));
+    $extension_info = [];
+
+    foreach ($acquia_extensions as $name => $extension) {
+      // Version is unset for dev versions.
+      $version = $extension['version'] ? $extension['version'] : $extension['core'];
+      $extension_info[$name]['version'] = $version;
+    }
+
+    $installed_modules = $this->extensionListModule->getAllInstalledInfo();
+    foreach ($acquia_extensions as $name => $extension) {
+      if (array_key_exists($name, $installed_modules)) {
+        $extension_info[$name]['status'] = 'enabled';
+      }
+      else {
+        $extension_info[$name]['status']  = 'disabled';
+      }
+    }
+
+    return $extension_info;
+  }
+
+  /**
+   * Creates an Amplitude event.
+   *
+   * @param string $type
+   *   The event type.
+   * @param array $properties
+   *   The event properties.
+   *
+   * @return \Drupal\acquia_telemetry\Event
+   *   An Amplitude event with basic info already populated.
+   */
+  protected function createEvent($type, $properties) {
+    $user_id = $this->getUserId();
+    $default_properties['extensions'] = $this->getExtensionInfo();
+    $default_properties['php']['version'] = phpversion();
+    $default_properties['drupal']['version'] = \Drupal::VERSION;
+    $properties = NestedArray::mergeDeep($default_properties, $properties);
+    $event = new Event($type, $user_id, $properties);
+
+    return $event;
+  }
+
+  /**
+   * Gets a unique ID for this application. "User ID" is an Amplitude term.
+   *
+   * @return string
+   */
+  protected function getUserId() {
+    return Crypt::hashBase64($this->config_factory->get('system.site')->get('uuid'));
+  }
+
+  /**
+   * Gets an array of all Acquia Drupal extensions.
+   *
+   * @return array
+   *   A flat array of all Acquia Drupal extensions.
+   */
+  public function getAcquiaExtensionNames(): array {
```

Comment:
I still think we should use ComponentDiscovery to do this. If we add a subcomponent (or a top-level one), we _will_ forget to update this obscure array.